lib/api/src/modules/url.ts

```diff
@@ -154,15 +137,35 @@ export const init: ModuleFn = ({ store, navigate, state, provider, fullAPI, ...r
     },
   };
 
-  const initModule = () => {
+  const initModule = async () => {
+    const { location, viewMode } = state;
+    const query = queryFromLocation(location);
+    const { selectedKind, selectedStory, path: queryPath } = query;
+
+    if (await fullAPI.showReleaseNotesOnLaunch()) {
+      setTimeout(() => navigate(`/settings/release-notes`, { replace: true }), 1);
+    } else if (selectedKind && selectedStory) {
+      const id = toId(selectedKind, selectedStory);
+      setTimeout(() => navigate(`/${viewMode}/${id}`, { replace: true }), 1);
+    } else if (selectedKind) {
+      // Create a "storyId" of the form `kind-sanitized--*`
+      const standInId = toId(selectedKind, 'star').replace(/star$/, '*');
+      setTimeout(() => navigate(`/${viewMode}/${standInId}`, { replace: true }), 1);
+    } else if (!queryPath || queryPath === '/') {
+      setTimeout(() => navigate(`/${viewMode}/*`, { replace: true }), 1);
+    } else if (Object.keys(query).length > 1) {
+      // remove other queries
+      setTimeout(() => navigate(`${queryPath}`, { replace: true }), 1);
+    }
+
     fullAPI.on(NAVIGATE_URL, (url: string, options: { [k: string]: any }) => {
       fullAPI.navigateUrl(url, options);
     });
   };
 
   return {
     api,
-    state: initialUrlSupport({ store, navigate, state, provider, fullAPI, ...rest }),
+    state: initialUrlSupport({ store, state, provider, fullAPI, ...rest }),
     init: initModule,
```

Comment:
What's the idea behind moving this? I can't say I fully understand why this is where it is (or what's up with the `setTimeout(..., 1)`s, but can you explain why it's moving?

Reply:
The whole `setTimeout` thing feels like a hack. TBH I am not sure why it is being done that way. I suspect it is some workaround for a timing issue related to postMessage events between the manager & preview. That said, I moved this stuff around because the `initialUrlSupport` function was triggering redirects _before_ all of the "modules" have been setup and therefore, I didn't have access to the state of the `release_notes` module to know if that required a redirect. By moving it, I could `await` the `release_notes` module & know whether or not we should redirect there.

Reply:
That said, I am slightly confused as there are a bunch of redirects happening & given what I had to do here:

https://github.com/storybookjs/storybook/pull/11360/files#r447366304

There doesn't seem to be a dedicated place to isolate this logic.

Reply:
@tmeasday I moved these redirects back for now and commented them out. I was planning to take another look at how it affects things once you've had the chance to look into removing them.

Reply:
My PR is merged now!

Reply:
Nice. I merged that in 👌